hazelcast/src/main/java/com/hazelcast/wan/WanReplicationService.java

```diff
@@ -96,6 +98,22 @@
      */
     void syncMap(String wanReplicationName, String targetGroupName, String mapName);
 
+    /**
+     * Initiate sync for a single map and specific collection of keys.
+     * NOTE: not supported on OS, only on EE
+     *
+     * @param wanReplicationName the name of the wan replication config
+     * @param targetGroupName    the group name on the target cluster
+     * @param mapName            the map name
+     * @param keys               keys to sync
+     * @param <K>                key type
+     * @throws UnsupportedOperationException if the operation is not supported (not EE)
+     * @throws InvalidConfigurationException if there is no WAN replication config for {@code wanReplicationName}
+     * @throws SyncFailedException           if there is a sync request in progress
+     */
+    <K> void syncMap(String wanReplicationName, String targetGroupName, String mapName,
```

Comment:
can't we access this behind `EnterpriseWanReplicationService` if this is EE only?

Reply:
Yes, that makes sense. Unfortunately we'll have to do some casting since `NodeEngineImpl#getWanReplicationService` returns the OS interface.

Reply:
I like your idea and I tried to implement it. I tried to separate OS functionality in `WanReplicationService` and EE functionality in a subinterface `EnterpriseWanReplicationService`. Unfortunately, it can't be done cleanly. There are other OS-side functionalities that invoke EE features and expect to work when invoked with EE. These include:
- REST API in `com.hazelcast.internal.ascii.rest.HttpPostCommandProcessor`
- mancenter tasks such as `ChangeWanStateRequest`
- stats sent to mancenter such as `TimedMemberStateFactory#createWanSyncState`

So I could move some methods to EE while some would need to stay in OS. Because this is then done inconsistently, I'd rather keep to this approach of exposing it all on OS and throwing `UnsupportedOperationException`.